### ISSUE TITLE
Upgrade markdownlint dependency to version 0.3.1

### ISSUE THREAD

**Issue by DavidAnson** [eslint/eslint#7589]
A new version of `markdownlint` is available. It includes the following improvements:

* 0.3.0 - More detailed error reporting with resultVersion, enhance MD010/MD012/MD036, fixes for MD027/MD029/MD030, include JSON schema, dependencies.
  * 0.3.1 - Fix regressions in MD032/MD038, update dependencies.

Opting into `resultVersion` 1 gives more helpful error messages:

```
bad.string: 1: MD018/no-missing-space-atx No space after hash on atx style header [Context: "#bad.string"]
```

No new violations are reported.

I will submit a pull request once this issue is accepted.

**Comment by not-an-aardvark** [comment]
Thanks!

> I will submit a pull request once this issue is accepted.

For what it's worth, we no longer require issues for most pull requests, so feel free to submit a pull request now if you'd like. (Since this is an internal build change, I doubt anyone will be opposed to the upgrade.)

**Comment by DavidAnson** [comment]
Thanks, I've created a PR just now.

FYI that the documentation may need to be updated to say that issues are optional for non-documentation changes:

> Make sure there is an issue for any pull request you send.
> The only exception is for documentation changes.

http://eslint.org/docs/developer-guide/contributing/pull-requests#step-5-double-check-your-submissiona-namestep5a

**Comment by kaicataldo** [comment]
Thanks - you're right, we should definitely update that
